### ISSUE TITLE
max-classes-per-file is ES6 only

### ISSUE THREAD

**Issue by ghost** [eslint/eslint#10469]
**What rule do you want to change?**
max-classes-per-file should not create issues if ES5 is selected as ecmaVersion and should be listed unser ECMAScrip6-specific-rules.

**Does this change cause the rule to produce more or fewer warnings?**
Fewer on <ES6.

**How will the change be implemented? (New option, new default behavior, etc.)?**
New behaviour.

**Please provide some example code that this change will affect:**
"class foo{}" is no valid ES5 code and the rule should not raise issues on ecmaVersion:5.

```js

```

**What does the rule currently do for this code?**
The rule creates issues for ES5 code.

**What will the rule do after it's changed?**
It creates only issues if >ES5 is selected in the config. And in the hompage rule documentation it is listed under ECMAScript6-specific rules.

**Comment by g-plane** [comment]
Classes is a feature in ES6+, not ES5.

**Comment by ilyavolodin** [comment]
I'm a bit confused. If you selected ES5, then parsing should fail whenever eslint will find `class` keyword. As such, this rule will never even get executed.
